src/cmd/sanity_test/restore/onedrive.go

```diff
@@ -66,15 +66,61 @@ func checkDriveRestoration(
 		// map itemID -> item size
 		fileSizes = make(map[string]int64)
 		// map itemID -> permission id -> []permission roles
-		folderPermissions         = make(map[string][]permissionInfo)
+		folderPermissions         = make(map[string][]PermissionInfo)
 		restoreFile               = make(map[string]int64)
-		restoredFolderPermissions = make(map[string][]permissionInfo)
+		restoredFolderPermissions = make(map[string][]PermissionInfo)
 	)
 
-	var restoreFolderID string
-
 	ctx = clues.Add(ctx, "drive_id", driveID, "drive_name", driveName)
 
+	restoreFolderID := PopulateDriveDetails(
+		ctx,
+		client,
+		driveID,
+		folderName,
+		dataFolder,
+		fileSizes,
+		folderPermissions,
+		startTime,
+	)
+
+	getRestoredDrive(ctx, client, driveID, restoreFolderID, restoreFile, restoredFolderPermissions, startTime)
+
+	checkRestoredDriveItemPermissions(
+		ctx,
+		service,
+		skipPermissionTest,
+		folderPermissions,
+		restoredFolderPermissions)
+
+	for fileName, expected := range fileSizes {
+		utils.LogAndPrint(ctx, "checking for file: %s", fileName)
+
+		got := restoreFile[fileName]
+
+		utils.Assert(
+			ctx,
+			func() bool { return expected == got },
+			fmt.Sprintf("different file size: %s", fileName),
+			expected,
+			got)
+	}
+
+	fmt.Println("Success")
+}
+
+func PopulateDriveDetails(
```

Comment:
Similarly here.  I think any code we share between packages should move to some third package.  That separates concerns between functionality and consumers, instead of having one file that contains both functionality _and_ a consumer, and other files that are consumer only.  It'll also provide a clear location in the long term for where to add new shared functionality.